domains/transfer-agreements/TransferAgreements.API/API.IntegrationTests/Controllers/TransferAgreementAuditsControllerTests.cs

```diff
@@ -0,0 +1,189 @@
+            DateTimeOffset.UtcNow.ToUnixTimeSeconds(),
+            DateTimeOffset.UtcNow.AddDays(1).ToUnixTimeSeconds(),
+            "12345678"
+        );
+
+        var createResponse = await authenticatedClient.PostAsJsonAsync("api/transfer-agreements", transferAgreement);
+        createResponse.EnsureSuccessStatusCode();
+
+        createResponse.StatusCode.Should().Be(HttpStatusCode.Created);
+
+        var createdTransferAgreement = await createResponse.Content.ReadFromJsonAsync<TransferAgreementDto>();
+        createdTransferAgreement.Should().NotBeNull();
+
+        var auditsResponse = await authenticatedClient.GetAsync($"api/audits/transfer-agreements/{createdTransferAgreement.Id}");
+        auditsResponse.EnsureSuccessStatusCode();
+
+        auditsResponse.StatusCode.Should().Be(HttpStatusCode.OK);
+
+        var audits = await auditsResponse.Content.ReadFromJsonAsync<TransferAgreementAuditsResponse>();
+        audits.Should().NotBeNull();
+        audits.Result.Should().HaveCount(1);
+    }
+
+    [Fact]
+    public async Task Edit_ShouldGenerateAudit_WhenTransferAgreementEndDateIsChanged()
+    {
+        var senderId = Guid.NewGuid();
+        var agreementId = Guid.NewGuid();
+
+        await SeedData(
+            new List<TransferAgreement>()
+            {
+                new()
+                {
+                    Id = agreementId,
+                    SenderId = senderId,
+                    StartDate = DateTimeOffset.UtcNow.AddDays(1),
+                    EndDate = DateTimeOffset.UtcNow.AddDays(10),
+                    ActorId = "actor1",
+                    ActorName = "Thomas Tesla",
+                    SenderName = "nrgi A/S",
+                    SenderTin = "44332211",
+                    ReceiverTin = "1122334"
+                }
+            });
+
+        var authenticatedClient = factory.CreateAuthenticatedClient(sub: senderId.ToString());
+
+        var newEndDate = DateTimeOffset.UtcNow.AddDays(15).ToUnixTimeSeconds();
+        var request = new EditTransferAgreementEndDate(newEndDate);
+
+        var response = await authenticatedClient.PatchAsync($"api/transfer-agreements/{agreementId}", JsonContent.Create(request));
+
+        response.StatusCode.Should().Be(HttpStatusCode.OK);
+
+        var updatedTransferAgreement = await response.Content.ReadFromJsonAsync<TransferAgreementDto>();
+        updatedTransferAgreement.Should().NotBeNull();
+        updatedTransferAgreement.EndDate.Should().Be(newEndDate);
+
+        var auditsResponse = await authenticatedClient.GetAsync($"api/audits/transfer-agreements/{agreementId}");
+        auditsResponse.EnsureSuccessStatusCode();
+
+        auditsResponse.StatusCode.Should().Be(HttpStatusCode.OK);
+
+        var audits = await auditsResponse.Content.ReadFromJsonAsync<TransferAgreementAuditsResponse>();
+        audits.Should().NotBeNull();
+        audits.Result.Should().HaveCount(2);
+        audits.Result.Last().EndDate.Should().Be(newEndDate);
```

Comment:
I will suggest that the entire content returned from the API is asserted and not just a single property. The test can become harder to read due to the many properties, but here we have previously used snapshot testing with the tool https://github.com/VerifyTests/Verify to make the tests more concise.

Reply:
I have refactored all tests, to now make use of the Verify library. Thanks for the suggestion :)